app/controllers/trips_controller.rb

```diff
@@ -0,0 +1,89 @@
+class TripsController < ApplicationController
+  def index
+    if params[:passenger_id]
+      @passenger = Passenger.find_by(id: params[:passenger_id])
+      @trips = @passenger.trips
+    else
+      @trips = Trip.all
+    end
+  end
+
+  def edit
+    @trip = Trip.find_by(id: params[:id])
+
+    if @trip.nil?
+      head :not_found
+      return
+    end
+  end
+
+  def update
+    @trip = Trip.find_by(id: params[:id])
+
+    if @trip.nil?
+      head :not_found
+      return
+    elsif @trip.update(trip_params)
+      redirect_to trip_path
```

Comment:
Just a suggestion, the trip is complete (rated etc), then you can make the driver available now.